### PR TITLE
feat(proposal): add np000 proposal

### PR DESCRIPTION
**Pull request by geoah**

### CODE REVIEW

File: docs/proposals/n000-proposals.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+### Consensus and Disagreement
+
+The goal of the proposal process is to reach general consensus about the outcome
+in a timely manner.
+
+If general consensus cannot be reached, the proposal review group decides the
+next step by reviewing and discussing the issue and reaching a consensus among
+themselves. If even consensus among the proposal review group cannot be reached
+(which should be exceedingly unusual), the arbiter (@geoah ) reviews the
```
Comment (jimeh) [inline docs/proposals/n000-proposals.md]:
```suggestion
(which should be exceedingly unusual), the arbiter (@geoah) reviews the
```

**Review by jimeh — CHANGES_REQUESTED** [review]

File: docs/proposals/n000-proposals.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+### Proposal Review
+
+A group of contributors will be holding “proposal review meetings” to review
+pending proposals.
+
+The principal goal of the review meeting is to make sure that proposals are
+receiving attention from the right people, by cc'ing relevant developers,
+raising important questions, pinging lapsed discussions, and generally trying to
+guide discussion toward agreement about the outcome. The discussion itself is
+expected to happen on the proposal's discussion page.
```
Comment (superdecimal) [inline docs/proposals/n000-proposals.md]: I am a bit confused as to if this is a video call or comments on the PR for the proposal.

Comment (superdecimal) [reply docs/proposals/n000-proposals.md]: The way I see this is a meeting that the notes are made on the proposal pr. Is this right?

Comment (jimeh) [reply docs/proposals/n000-proposals.md]: I guess maybe this is intentionally vague as to not be restrictive? But maybe some suggestions could be mentioned like where/how this happens, on a PR adding the proposal document, etc.

File: docs/proposals/n000-proposals.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+The proposal process itself is influenced by
+[golang's](https://github.com/golang/proposal#design-documents) and
+[ethereum's](https://eips.ethereum.org/EIPS/eip-1)
+proposal processes.
+
+### Index example
+
+* Objects
+  * np001 - Hinted objects
```
Comment (superdecimal) [inline docs/proposals/n000-proposals.md]: We could have another level of numbers to link various proposals together. ie `np001-01 - Hinted Objects v2`

File: docs/proposals/n000-proposals.md (hunk 2 above)
Comment (geoah) [reply docs/proposals/n000-proposals.md]: reworded this section a bit.
all discussion should be happening on the discussions page.
meeting is to make sure that we are dealing with proposals.

File: docs/proposals/n000-proposals.md (hunk 3 above)
Comment (geoah) [reply docs/proposals/n000-proposals.md]: Hm, interesting point, I think we can deal with this as we move forward. I'll look into how RFCs and other people deal with this for now.

**Pull request merged**